ui/src/views/Scopes/ScopesetComparison/index.jsx

```diff
@@ -49,6 +50,23 @@ export default class ScopesetComparison extends Component {
     this.setState({ scopeTextB });
   };
 
+  componentDidMount() {
+    const query = parse(this.props.location.search.slice(1));
+    const { scopesA, scopesB } = query;
+
+    if (scopesA && scopesB) {
+      const scopesetDiff = this.getScopesetDiff(scopesA, scopesB);
+      const cellColors = this.getCellColors(scopesetDiff);
+
+      this.setState(() => ({
+        scopeTextA: scopesA.join('\n'),
+        scopeTextB: scopesB.join('\n'),
+        scopesetDiff,
+        cellColors,
+      }));
+    }
+  }
```

Comment:
The React [docs](https://reactjs.org/docs/react-component.html#componentdidmount) don't suggest calling `this.setState` in `componentDidMount` for performance reasons. We can avoid this by having this logic in the constructor.

```jsx
constructor(props) {
  super(props);
  
  // ...
  
  this.state = {
    // ...
  };
}
```

_Note: We might not be able to call `this.<something>` inside the constructor so we would need to extract things like `getScopesetDiff` and  `getCellColors` to be outside the class. Putting them after the imports lines of this file sounds good to me._

Reply:
I have removed the need for the `componentDidMount()` lifecycle method and replaced it with the constructor method.